### PR TITLE
feat: support package debian

### PR DESCRIPTION
**Pull request by soulbird**

### CODE REVIEW

**Comment by soulbird** [comment]
> Do you have some successful CI processes running in your own repository?

I changed the CI trigger condition, and tested the packaging and uploading, and it was ok.

File: utils/publish-deb.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    find "${1}" -type f -name "apsix_*.deb" \
+        -exec echo "upload : {}" \; \
+        -exec sh -c 'file=$(basename {}); \
+                    coscli -e "${VAR_COS_ENDPOINT}" cp {} "cos://${2}/packages/${3}/pool/main/a/apisix/${file}"' \;
+
+    find "${1}" -type f -name "apsix-base*.deb" \
+        -exec echo "upload : {}" \; \
+        -exec sh -c 'file=$(basename {}); \
+                    coscli -e "${VAR_COS_ENDPOINT}" cp {} "cos://${2}/packages/${3}/pool/main/a/apisix-base/${file}"' \;
+}
```
Comment (tzssangglass) [inline utils/publish-deb.sh:52]: we don't need to distinguish apisix or apisix-base here?

just find `*.deb` in work dir and upload is ok?

also, we should add `--part-size 1000` when upload.

Comment (soulbird) [reply utils/publish-deb.sh:52]: This is the directory structure where the debian repositories are installed to create the path, although it's still just a file

File: .github/workflows/publish-deb.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          echo "##[set-output name=version;]$(echo ${GITHUB_REF#refs/tags/})"
+
+      - name: Build apisix deb Package
+        if: ${{ startsWith(steps.tag_type.outputs.version, 'apisix/') }}
+        env:
+          APISIX_TAG_VERSION: ${{ steps.tag_env.outputs.version}}
+        run: |
+          # build apisix
+          make package type=deb app=apisix openresty=apisix-base checkout=${APISIX_TAG_VERSION} version=${APISIX_TAG_VERSION} image_base=debian image_tag=bullseye-slim
+          mv ./output/apisix_${APISIX_TAG_VERSION}-0~debianbullseye-slim_amd64.deb ${VAR_DEB_WORKBENCH_DIR}
```
Comment (spacewander) [inline .github/workflows/publish-deb.yml:51]: Could we move `debianbullseye-slim` into a variable?

File: utils/publish-deb.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+func_repo_upload() {
+    # ${1} - local path
+    # ${2} - bucket name
+    # ${3} - COS path
+    find "${1}" -type f -name "apsix_*.deb" \
+        -exec echo "upload : {}" \; \
+        -exec sh -c 'file=$(basename {}); \
+                    coscli -e "${VAR_COS_ENDPOINT}" cp {} --part-size 1000 "cos://${2}/packages/${3}/pool/main/a/apisix/${file}"' \;
```
Comment (tzssangglass) [inline utils/publish-deb.sh:46]: what is `cp {}` used?

**Review by membphis — APPROVED** [review]

File: .github/workflows/publish-deb.yml (hunk 2 above)
Comment (soulbird) [reply .github/workflows/publish-deb.yml:51]: Just add two env vars:
`
env:
   PACKAGE_OS: debian
   PACKAGE_OS_RELEASE: bullseye-slim
`
what do you think?

Comment (soulbird) [reply .github/workflows/publish-deb.yml:51]: @spacewander The definition of the package name is scattered, there is currently no unified function to define it, maybe we need another PR to solve it.

**Review by tzssangglass — APPROVED** [review]

File: .github/workflows/publish-deb.yml (hunk 2 above)
Comment (spacewander) [reply .github/workflows/publish-deb.yml:51]: OK

**Review by spacewander — APPROVED** [review]

**Pull request merged**